app/js/components/service_form.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+let serviceForm = new function($) {
+
+    let $privacyQuestionsEnabledToggle;
+    let $privacyQuestionsStatusContainer;
+    let $contractSignedToggle;
+    let $contractSignedContainer;
+
+    /**
+     * Initialize events
+     * @param privacyQuestionToggle
+     * @param privacyQuestionContainer
+     * @param contractSignedToggle
+     * @param contractSignedContainer
+     */
+    function initializeEventHandlers(
+        privacyQuestionToggle,
+        privacyQuestionContainer,
+        contractSignedToggle,
+        contractSignedContainer
+    ) {
+        $privacyQuestionsEnabledToggle = $(privacyQuestionToggle);
+        $privacyQuestionsStatusContainer = $(privacyQuestionContainer);
+        $contractSignedToggle = $(contractSignedToggle);
+        $contractSignedContainer = $(contractSignedContainer);
+
+        // Init the privacy questions answered toggle
+        initTogglePrivacyQuestionsAnsweredStatusField();
+
+        // Init the contract signed toggle
+        initToggleContractSignedStatusField();
+    }
+
+    /**
+     * Disable the contract signed field when the type of service is for an institute
+     */
+    function initToggleContractSignedStatusField()
+    {
+        let toggleContractSigned = function () {
+            let serviceTypeValue = $contractSignedToggle.find(':checked').val();
+            if (serviceTypeValue === "institute") {
+                $contractSignedContainer.parent().show();
+            } else {
+                $contractSignedContainer.parent().hide();
+            }
+        };
+
+        $contractSignedToggle.on('change', toggleContractSigned);
+        toggleContractSigned();
+    }
+
+    /**
+     * Disable the privacy questions answered field when the privact question enabled checkbox is unchecked
+     */
+    function initTogglePrivacyQuestionsAnsweredStatusField()
+    {
+        let togglePrivacyQuestions = function () {
+            if ($privacyQuestionsEnabledToggle.is(':checked')) {
+                $privacyQuestionsStatusContainer.parent().show();
```

Comment:
You removed the show animation (fadein 200ms). Is this something we really want? In my opinion, some animation is beneficial for a good user experience.

Reply:
I removed it to be able to test the logic without waiting till the animation has ended. I've now added a class, so I could check for that `class` in the tests.